.pre-commit-config.yaml

```diff
@@ -0,0 +1,18 @@
+repos:
+  - repo: https://github.com/PyCQA/flake8
+    rev: '6.0.0'
```

Comment:
Can we use latest by default? (also mean that CI need to force fresh download for each run)
It will force us to use latest standard and avoid upgrading only once by ten years

Reply:
In theory you can, but it will warn you every time that you should lock it. If they change the options some day or they drop support for old packages we are using you will randomly have tests that fail. Their behaviour should be stable, no?

Reply:
Yeah I understand, but IMHO, it's the difference between theory and practice
We use this way with dev tools (pytest, test requirements, etc..) and it avoid a lot of technical debt with time. Because we both know that nobody will take 1-2 days (including review/ci/merge) to update all versions from all repositories each X time.
But, I'm just asking...  I don't have objection to pin version. If it can avoid an annoying warning, it's fine.

Reply:
I suppose. Less explicit, but less maintenance until they all break at the same time. It will do a big warning every time it's run though. Especially locally it would b quite annoying. I've always automated my repos with tools like dependabot that automatically make PRs to upgrade stuff. Given the number of repos we have it may be something to look into some day.